ksql-parser/src/main/java/io/confluent/ksql/parser/json/ExpressionSerializer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser.json;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import io.confluent.ksql.execution.expression.formatter.ExpressionFormatter;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.util.IdentifierUtil;
+import java.io.IOException;
+
+class ExpressionSerializer extends JsonSerializer<Expression> {
```

Comment:
Can we not have a single serializer for any `Node` type?  Why do we need different ones for each type, given they're all doing the same thing, i.e. converting to a string?

Reply:
I don't think it makes sense to use `toString` to serialize these. There's no general understanding that `Node.toString` serializes to SQL. For example, `GroupBy.toString` doesn't serialize to SQL, and its likely a developer adding a node won't write toString this way. I think using `SqlFormatter/ExpressionFormatter` makes sense. Currently SqlFormatter can't handle arbitrary `AstNode` instances. I can extend that, and then change the window expression serializer to an AstNode serializer - would rather leave that for a different change though.